os/kernel/debug/sysdbg.c

```diff
@@ -0,0 +1,941 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/***************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <errno.h>
+#include <tinyara/config.h>
+#include <tinyara/arch.h>
+#include <tinyara/kmalloc.h>
+#include <tinyara/debug/sysdbg.h>
+#include "sched/sched.h"
+
+/* Kindly update the debug_version as and when changes done in struct
+ * sysdbg_struct as there will be dependency on T32 scripts and
+ * python script to handle the structure */
+volatile int g_debug_version = 1;
```

Comment:
What's the usages of this? Now, this is not called by anyplaces.

Reply:
This will be used in future by python script to parse the debug information based on the version